avocado_vt/test.py

```diff
@@ -202,6 +202,9 @@ def runTest(self):
 
     def _runTest(self):
         params = self.params
+        if params.get("test_pre_hook"):
+            self.log.info(f"Executing VT test pre-hook from {params['test_pre_hook']}")
+            exec(open(params["test_pre_hook"]).read())
```

Comment:
Hi @pevogam I am not quite understand how this should work. Can you give me an example how the `params["test_pre_hook"]` value should look like? Also I have a feeling that this code might be possible to move to the `env_process.preprocess` method and copy the hooks from `params` to the preprocess_vm_off_hook and other variables. What do you think?

Reply:
What I used for testing here was a path to a small customization script as you could probably guess from `exec(open(params["test_pre_hook"]).read())`. This is where I could do all of the steps like:

1) Changing logging levels for some external libraries like logging.getLogger("extlib").setLevel(99)
2) Inserting some special utility paths like `sys.path.insert(1, utility_path)`.
3) Setting multiple env processing hooks like
```python
env_process.preprocess_vm_off_hook = off_state(ss.get_states)
env_process.preprocess_vm_on_hook = on_state(ss.get_states)
env_process.postprocess_vm_on_hook = on_state(ss.set_states)
env_process.postprocess_vm_off_hook = off_state(ss.set_states)
```
As you can notice, we might use some `params` keys for the env process hooks but this is just a minor part of what we actually need to do in order to customize the runner behavior.

Reply:
My only remark is regarding not explicitly closing the file, which puts the burden on the GC and in some rare cases can lead to issues (such as running ou of file descriptors or preventing unmount operations).

Reply:
I would have preferred the one liner knowing this since there isn't much danger of unmounting in this stage of the code and GC works just fine (and it was the main recommended way on some source forums) but we cannot sacrifice some better patterns for the sake of prettier syntax and API limitations so I went ahead with a context manager.